### DISCUSSION TITLE
how can I bind to my own domain? TcpListener::bind("own_domain")

### DISCUSSION THREAD

**Discussion by AleshaKaramazov** [tokio-rs/tokio#6824] [Q&A]
Recently I got my own domain (let's say "example.online") and I had a desire to write a server part for my application with
`let listener = TcpListener::bind("example.online").await.unwrap();`
but instead of bind it gives me an error:
`called `Result::unwrap()` on an `Err` value: Os { code: 49, kind: AddrNotAvailable, message: "Can't assign requested address" }`
Is it possible to somehow fix this? I have a hosting and a DNS server, the сite itself works fine if connect to it on the Internet, but for some reason the binding does not work

**Comment by Darksonn** [comment]
You do not bind to a domain. You can either bind to a specific interface, or to all interfaces. Usually you only have two interfaces: localhost and your internet connection, but if you're connected over both ethernet and wifi you might have three interfaces. Binding to `127.0.0.1` only accepts connections from the local machine. Binding to `0.0.0.0` accepts connections from anywhere. Most likely you should just bind to `0.0.0.0`.

**Reply by AleshaKaramazov** [reply]
but 0.0.0.0 is also a local address, and I need it not to be local. my domain has an ip server too, but I can’t bind to it

**Reply by Darksonn** [reply]
Are you trying to prevent connections from localhost? Otherwise I don't understand. If you bind to 0.0.0.0, then that accepts connections from anywhere, local or not.

**Reply by AleshaKaramazov** [reply]
What do I need to enter in the address bar to get to the port 0.0.0.0 page?

**Reply by Darksonn** [reply]
You can enter any IP address or domain that refers to your machine. As long as the port matches, it should work. For example, if you bind to `0.0.0.0:8123`, then the local machine can reach it via `127.0.0.1:8123`. Similarly, if the server has IP address `192.168.0.17` on the local network, then other machines on the local network can reach it at `192.168.0.17:8123` (unless your router blocks the connection).
